marathon_acme/server.py

```diff
@@ -26,19 +27,18 @@ def set_health_handler(self, health_handler):
         """
         self.health_handler = health_handler
 
-    def run(self, host, port, log_file=None):
+    def listen(self, host, port, clock):
         """
         Run the server, i.e. start listening for requests on the given host and
         port.
 
-        :param host:
-            The address to the interface to listen on.
-        :param port:
-            The port to bind to.
-        :param log_file:
-            The file to write logs to.
+        :param host: The address for the interface to listen on.
+        :param port: The port to bind to.
+        :param clock: The ``IReactorTCP`` to use.
+        :return: An object that provides ``IListeningPort``.
         """
-        self.app.run(host, port, log_file)
+        site = Site(self.app.resource())
+        return clock.listenTCP(port, site, interface=host)
```

Comment:
It seemed like more work? We're not really interested in things other than TCP I think. I went back and looked at Consular and this is roughly what it does.